Tests/GRPCCoreTests/MetadataTests.swift

```diff
@@ -252,4 +252,52 @@ struct MetadataTests {
       #expect(self.metadata == ["key1": "value1", "key3": "value1"])
     }
   }
+
+  @Suite("Merge")
+  struct Merge {
+    var metadata: Metadata = [
+      "key1": "value1-1",
+      "key2": "value2",
+      "key3": "value3",
+    ]
+    var otherMetadata: Metadata = [
+      "key4": "value4",
+      "key5": "value5",
+    ]
+
+    @Test("Where key is already present with a different value")
+    mutating func mergeWhereKeyIsAlreadyPresentWithDifferentValue() async throws {
+      self.otherMetadata.addString("value1-2", forKey: "key1")
+      self.metadata.merge(otherMetadata)
+
+      #expect(self.metadata == [
+        "key1": "value1-1",
+        "key2": "value2",
+        "key3": "value3",
+        "key4": "value4",
+        "key5": "value5",
+        "key1": "value1-2",
+      ])
+    }
+
+    @Test("Where key is already present with same value")
+    mutating func mergeWhereKeyIsAlreadyPresentWithSameValue() async throws {
+      self.otherMetadata.addString("value1-1", forKey: "key1")
+      self.metadata.merge(otherMetadata)
+
+      #expect(self.metadata == [
+        "key1": "value1-1",
+        "key2": "value2",
+        "key3": "value3",
+        "key4": "value4",
+        "key5": "value5",
+        "key1": "value1-1",
```

Comment:
I don't think this pair should actually be here, but this is a result of how `Metadata` was implemented (see the note in the PR description). I think we should consider changing this.

Reply:
I don't think we should: there's no requirement for key/value pairs of metadata to be unique.

I think `merge` is a misleading name here (you can merge values into a dictionary because it _does_ have a requirement that keys are unique). In this case a more appropriate spelling would be `add(contentsOf metadata: Metadata)` (we should also add an `add(contentsOf other: some Sequence<Element>)` while we're at it).

Reply:
The `some Sequence<Element>` variant is enough to be able to pass a `Metadata`. Should we still have the `Metadata` overload for optimisation purposes or is it unnecessary?

Reply:
Indeed it saves an allocation, I've added it.